pkg/tenant/projects.go

```diff
@@ -122,16 +132,17 @@ func (s *Server) TenantProjectCreate(c *gin.Context) {
 		Name:     project.Name,
 	}
 
-	// Add project to the database and return a 500 response if it cannot be added.
-	if err = db.CreateTenantProject(c.Request.Context(), tproject); err != nil {
-		log.Error().Err(err).Msg("could not create tenant project in the database")
-		c.JSON(http.StatusInternalServerError, api.ErrorResponse("could not add tenant project"))
+	// Create the project in the database and register it with Quarterdeck.
+	// TODO: Distinguish between trtl errors and quarterdeck errors.
+	if err = s.createProject(ctx, tproject); err != nil {
+		log.Error().Err(err).Msg("could not create project")
+		c.JSON(qd.ErrorStatus(err), api.ErrorResponse("could not create project"))
```

Comment:
If this is a `trtl` error what will `qd.ErrorStatus(err)` do - does it return a 500?

Reply:
I will return a 500. I was thinking we might want to have a struct similar to `StatusError` either in the tenant db package or in trtl so that we could switch on on the type here and return a more useful status code.

Reply:
Makes sense to me - would you mind creating a follow on story for that?